### PR TITLE
12.0 mig account payment mode

### PR DESCRIPTION
**Pull request by mourad-ehm**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Check linter errors and squash a bit commit history (at least Transbot ones).

File: account_payment_mode/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
```
Comment (ValentinVinagre) [inline account_payment_mode/__init__.py]: They are not necessary.

File: account_payment_mode/models/account_payment_method.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# © 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
```
Comment (ValentinVinagre) [inline account_payment_mode/models/account_payment_method.py:4]: I think the '_' is missing

File: account_payment_mode/models/account_payment_mode.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    def _onchange_company_id(self):
+        for mode in self:
+            mode.variable_journal_ids = False
+            mode.fixed_journal_id = False
+
+    @api.constrains(
+        'bank_account_link', 'fixed_journal_id', 'payment_method_id')
+    def bank_account_link_constrains(self):
+        for mode in self:
+            if mode.bank_account_link == 'fixed':
```
Comment (ValentinVinagre) [inline account_payment_mode/models/account_payment_mode.py]: I would put this if up
´for mode in self.filtered(lambda x: x.bank_account_link == 'fixed'):´

File: account_payment_mode/models/account_payment_mode.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                                "of the debit methods of the fixed bank "
+                                "journal '%s'") % (
+                                    mode.name,
+                                    mode.payment_method_id.name,
+                                    mode.fixed_journal_id.name))
+
+    @api.constrains('company_id', 'fixed_journal_id')
+    def company_id_fixed_journal_id_constrains(self):
+        for mode in self:
+            if mode.fixed_journal_id and mode.company_id != \
```
Comment (ValentinVinagre) [inline account_payment_mode/models/account_payment_mode.py]: I would put this if up
for mode in self.filtered(
            lambda x: x.fixed_journal_id and x.company_id != x.fixed_journal_id.company_id
        ):

File: account_payment_mode/models/res_partner_bank.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# © 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
```
Comment (ValentinVinagre) [inline account_payment_mode/models/res_partner_bank.py:4]: I think the '_' is missing

File: account_payment_mode/security/account_payment_mode.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
```
Comment (ValentinVinagre) [inline account_payment_mode/security/account_payment_mode.xml:2]: I think the © is missing

File: account_payment_mode/views/account_journal.xml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (ValentinVinagre) [inline account_payment_mode/views/account_journal.xml:2]: I think the © is missing

**Review by ValentinVinagre — COMMENTED** [review]
I have commented several things in case you consider it appropriate to make the changes.

File: account_payment_mode/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# © 2016 Akretion (<https://www.akretion.com>).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Account Payment Mode',
+    'version': '11.0.1.0.1',
```
Comment (sergio-teruel) [inline account_payment_mode/__manifest__.py]:
```suggestion
    'version': '12.0.1.0.0',
```

File: account_payment_mode/migrations/11.0.1.0.1/pre-migration.py [hunk 9]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+def migrate(cr, version):
+    if not version:
+        return
+
+    cr.execute('ALTER TABLE res_partner_bank DROP COLUMN acc_type')
```
Comment (sergio-teruel) [inline account_payment_mode/migrations/11.0.1.0.1/pre-migration.py]: Remove this file

File: account_payment_mode/__init__.py (hunk 1 above)
Comment (sergio-teruel) [reply account_payment_mode/__init__.py]: You can remove all utf-8 with this command:
[https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0#tasks-to-do-in-the-migration](url)

File: account_payment_mode/models/account_payment_method.py (hunk 2 above)
Comment (sergio-teruel) [reply account_payment_mode/models/account_payment_method.py:4]: Is not necessary for _sql_constraint messages

File: account_payment_mode/models/account_payment_mode.py [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    # I dropped sale_ok and purchase_ok fields, because it is replaced by
+    # payment_type = 'inbound' or 'outbound'
+    # In fact, with the new v9 datamodel, you MUST create 2 payment modes
+    # for wire transfer : one for wire transfer from your customers (inbound)
+    # and one for wire transfer to your suppliers (outbound)
+    note = fields.Text(string="Note", translate=True)
+
+    @api.onchange('company_id')
+    def _onchange_company_id(self):
+        for mode in self:
```
Comment (sergio-teruel) [inline account_payment_mode/models/account_payment_mode.py]: loop is not necessary for onchange methods.

File: account_payment_mode/models/res_partner_bank.py (hunk 5 above)
Comment (sergio-teruel) [reply account_payment_mode/models/res_partner_bank.py:4]: @ValentinVinagre _ not ussed

**Review by sergio-teruel — CHANGES_REQUESTED** [review]
little comments

**Comment by HaraldPanten** [comment]
@mourad-ehm 

Are you taking care of this PR?

File: account_payment_mode/models/account_payment_mode.py (hunk 3 above)
Comment (mourad-ehm) [reply account_payment_mode/models/account_payment_mode.py]: Ok modified in the last commit

File: account_payment_mode/models/account_payment_mode.py (hunk 4 above)
Comment (mourad-ehm) [reply account_payment_mode/models/account_payment_mode.py]: Ok modified in the last commit

File: account_payment_mode/migrations/11.0.1.0.1/pre-migration.py (hunk 9 above)
Comment (mourad-ehm) [reply account_payment_mode/migrations/11.0.1.0.1/pre-migration.py]: Ok modified in the last commit

**Comment by pedrobaeza** [comment]
I have fixed remaining things and everything is OK, so I'm merging this one and I'll continue with the rest of this repo for finishing things.

**Comment by HaraldPanten** [comment]
THX @pedrobaeza

**Pull request merged**